### ISSUE TITLE
Three-command script (change workspace, open app, sleep 2s) opens in the wrong workspace only when called with Sway keybinding (unless removing sleep)

### ISSUE THREAD

**Issue by sebastiancarlos** [swaywm/sway#7864]
- **Sway Version: 1.8.1**

- **Debug Log:**
  - [gist](https://gist.github.com/sebastiancarlos/f08c05401a89f488f138fe8d51e3eae6)
  - Note that the script is run twice (once from the command line, where it works, and then with the keybinding)

- **Configuration File:**
  - The default configuration.
  - The only addition is this line: `bindsym $mod+z exec buggy-command`

### **Description:**
The faulty script, `buggy-command` is available in the path and contains. 

```bash
#!/usr/bin/env bash
swaymsg workspace 2
/usr/bin/foot &
/usr/bin/sleep 6
```

It is meant to be executed from a different workspace. Let's suppose it is executed from workspace 1.

If I run it with bash (`$ buggy-command`), it correctly switches to workspace 2, opens foot, and then waits a few seconds before concluding.

If I run it with the defined keybinding (`$mod+z`), then it switches to workspace 2, but opens foot on workspace 1.

Oddly enough, removing the sleep command fixes the issue. Here, it works both from the shell and from the keybinding:

```bash
#!/usr/bin/env bash
swaymsg workspace 2
/usr/bin/foot &
```

Naturally, the `sleep` at the end is a requirement of mine, as this is a simplified reproduction of a script which is meant to open an app and then wait until the app is open before returning control, which in turns enables fast chaining of commands to open multiple applications.

Note that the following variation does *not* fix the issue either:

```bash
#!/usr/bin/env bash
swaymsg workspace 2
/usr/bin/sleep 6 # new line
/usr/bin/foot &
/usr/bin/sleep 6
```

In other words, it's not a matter of waiting for the workspace switch to take effect. The issue indeed, and oddly, seems to be the `sleep` ***after*** the `foot` line.

**Comment by kennylevinsen** [comment]
It can be reproduced with just:

```
#!/usr/bin/env bash
swaymsg workspace 2
/usr/bin/foot
```

The issue is that the *outer* `swaymsg exec buggy-command` sets a startup token, which registers that the command was run from workspace 1. This token is not used until foot starts, which consumes the token and puts it on workspace 1.

There is not currently a way to avoid registering a startup token, but if you wrap foot in a new swaymsg exec it will get its own startup token assigned to workspace 2.
